docs/overview/release-lifecycle.md

```diff
@@ -1,151 +1,176 @@
+  at <https://chat.zulip.org> runs the `chat.zulip.org` branch. It's upgraded
+  to `main` several times a week. We also often "test deploy" changes not yet in
+  `main` to this branch, to facilitate design feedback.
+- [Zulip Cloud](https://zulip.com) runs the `zulip-cloud-current` branch plus
+  some cherry-picked changes. This branch is usually delayed by one to two weeks
+  from `main` to allow for recent changes to be validated further prior to being
+  deployed to customers.
+- You can also run [a fork of Zulip][fork-zulip].
 
-### Compatibility and upgrading
+[upgrade-from-git]: ../production/upgrade.md#upgrading-from-a-git-repository
 
-A Zulip design goal is for there never to be a reason to run an old
-version of Zulip. We work extremely hard to make sure Zulip is stable
-for self-hosters, has no regressions, and that the [Zulip upgrade
-process](../production/upgrade.md) Just Works.
+### What version am I running?
 
-The Zulip server and client apps are all carefully engineered to
-ensure compatibility with old versions. In particular:
+The Zulip web app displays the current server version [in the gear
+menu](https://zulip.com/help/view-zulip-version); it's also available [via the
+API](https://zulip.com/api/get-server-settings).
 
-- The Zulip mobile and desktop apps maintain backwards-compatibility
-  code to support any Zulip server version from the last 18 months.
-- Zulip maintains an [API changelog](https://zulip.com/api/changelog)
-  detailing all changes to the API to make it easy for client
-  developers to do this correctly.
-- The Zulip server preserves backwards-compatibility in its API to
-  support versions of the mobile and desktop apps released in roughly
-  the last year. Because these clients auto-update, generally there
-  are only a handful of active clients left by the time we desupport a
-  version.
+### Versioned documentation
 
-As a result, we generally do not backport changes to previous stable
-release series except in rare cases involving a security issue or
-critical bug just after publishing a major release.
+To make sure you can access documentation for your server version, [the help
+center](https://zulip.com/help/), [API documentation](https://zulip.com/api/),
+and [integrations documentation](https://zulip.com/integrations/) are
+distributed with the Zulip server itself (e.g.,
+`https://zulip.example.com/help/`).
 
-[upgrade-from-git]: ../production/upgrade.md#upgrading-from-a-git-repository
+This ReadTheDocs documentation has a widget in the top left corner
+that lets you view the documentation for other versions.
 
-### Security releases
+## Client apps
 
-When we discover a security issue in Zulip, we publish a security and
-bug fix release, transparently documenting the issue(s) using the
-industry-standard [CVE advisory process](https://cve.mitre.org/).
+Zulip's official client apps support all Zulip server versions (and
+Git commits) released in the **previous 18 months**.
```

Comment:
The git commits framing feels a bit confusing -- I don't think we want to be saying if we backport a commit to the 7.x branch to fix an upgrading problem, that we'll support that commit for 18 months.

Reply:
This is copied over from what it used to say. Should we drop that parenthetical?